### ISSUE TITLE
Support for cert-manager secret format in cacerts

### ISSUE THREAD

**Issue by nmnellis** [istio/istio#38528]
(This is used to request new product features, please visit <https://discuss.istio.io> for questions on using Istio)

**Describe the feature request**

I would like to use cert-manager to create the cacerts intermediate certificate secret that gets used by Istio to issue workload certificates. 

Currently i have to manually copy the secrets locally to put them into Istios cacert format. The secret contains all the necessary information but Istio cannot read it.

## Steps to create istio intemediate and copy to cacerts

1. Generate `root-ca`  and copt to kubernetes from the `1-certificates` folder

```sh
mkdir ./certs
openssl req -new -newkey rsa:4096 -x509 -sha256 \
        -days 3650 -nodes -out ./certs/root-ca.crt -keyout ./certs/root-ca.key \
        -config 1-certificates/root-ca.conf

kubectl create namespace cert-manager
kubectl create secret generic root-ca \
  --from-file=tls.key=./certs/root-ca.key \
  --from-file=tls.crt=./certs/root-ca.crt \
  --namespace cert-manager
```

2. Install cert-manager

```sh
kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.8.0/cert-manager.yaml

# wait for cert manager pod
kubectl wait --for=condition=ready pod -l app=cert-manager -n cert-manager
kubectl wait --for=condition=ready pod -l app=webhook -n cert-manager
```

3. Load `root-ca` into cert-manager

```sh
kubectl apply -f- <<EOF
apiVersion: cert-manager.io/v1
kind: Issuer
metadata:
  name: root-ca
  namespace: cert-manager
spec:
  ca:
    secretName: root-ca
EOF
```

4. Create an intermediate for Istio
```sh
kubectl apply -f- <<EOF
apiVersion: cert-manager.io/v1
kind: Certificate
metadata:
  name: istio-cacerts
  namespace: cert-manager
spec:
  secretName: istio-cacerts
  duration: 720h # 30d
  renewBefore: 360h # 15d
  commonName: istio.solo.io
  isCA: true
  usages:
    - digital signature
    - key encipherment
    - cert sign
  dnsNames:
    - istio.solo.io
  # Issuer references are always required.
  issuerRef:
    kind: Issuer
    name: root-ca
---
EOF
```
5. Download the cert-manager secret and upload to istio-system in correct format
```
kubectl get secret istio-cacerts -n istio-system -o json | jq '.data."tls.crt"' -r | base64 --decode > ./certs/ca-cert.pem
kubectl get secret istio-cacerts -n istio-system -o json | jq '.data."tls.key"' -r | base64 --decode > ./certs/ca-key.pem
kubectl get secret istio-cacerts -n istio-system -o json | jq '.data."ca.crt"' -r | base64 --decode > ./certs/root-cert.pem
kubectl get secret istio-cacerts -n istio-system -o json | jq '.data."tls.crt"' -r | base64 --decode > ./certs/cert-chain.pem
kubectl get secret istio-cacerts -n istio-system -o json | jq '.data."ca.crt"' -r | base64 --decode >> ./certs/cert-chain.pem

# create the istio secrets from the download files
kubectl create secret generic cacerts -n istio-system \
      --from-file=./certs/ca-cert.pem \
      --from-file=./certs/ca-key.pem \
      --from-file=./certs/root-cert.pem \
      --from-file=./certs/cert-chain.pem
```

**Affected product area (please put an X in all that apply)**

[ ] Docs
[X] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[X] Security
[ ] Test and Release
[ X] User Experience
[ ] Developer Infrastructure

**Affected features (please put an X in all that apply)**

[X] Multi Cluster
[X] Virtual Machine
[ ] Multi Control Plane

**Additional context**

**Comment by hzxuzhonghu** [comment]
Does not cert-manager allow specifying secret format?  I do not think istio should comply with all format, we should make others comply with us to build an ecosystem.

**Comment by nmnellis** [comment]
Let me rephrase the issue. Cert manager generates the secrets in the kubernetes `tls` secret type, a well known standard within Kubernetes. Istio should support reading certs in this format as it is the most common. 

```yaml
+ kubectl get secret -n cert-manager istio-cacerts -o yaml
apiVersion: v1
data:
  ca.crt: <ca>
  tls.crt: <cert>
  tls.key: <key>
kind: Secret
metadata:
  annotations:
    cert-manager.io/alt-names: istio.solo.io
    cert-manager.io/certificate-name: istio-cacerts
    cert-manager.io/common-name: istio.solo.io
    cert-manager.io/ip-sans: ""
    cert-manager.io/issuer-group: ""
    cert-manager.io/issuer-kind: ClusterIssuer
    cert-manager.io/issuer-name: root-ca
    cert-manager.io/uri-sans: ""
  creationTimestamp: "2022-04-22T15:15:09Z"
  name: istio-cacerts
  namespace: cert-manager
  resourceVersion: "3936911"
  uid: 472440bb-888e-4e77-bbfe-0dc9a90d2d2a
type: kubernetes.io/tls
```

**Comment by nmnellis** [comment]
this is complete

**Comment by eigood** [comment]
Which version has this fixed/implemented?  I'm running 1.14.1, and have this issue using gateway-api, and cert-manager+letsencrypt.  The secret type is tls, which isn't recognized.

Will this fix be backported to the 1.14 branch?

**Comment by nmnellis** [comment]
this made it into 1.14.2

**Comment by eigood** [comment]
Thank you!

**Comment by eigood** [comment]
I've got 1.14.3 installed now, but the Gateway.gateway.networking.k8s.io/v1beta1, running docker.io/istio/proxyv2:1.14.3, has this:

<pre>
==
      Last Transition Time:  2022-08-24T20:17:55Z
      Message:               invalid certificate reference core/Secret/um-wildcard-cert., only secret is allowed
      Observed Generation:   1
      Reason:                InvalidCertificateRef
      Status:                False
      Type:                  ResolvedRefs
    Name:                    https-wildcard
==
root@um-adam:~# kubectl get secret um-wildcard-cert 
NAME               TYPE                DATA   AGE
um-wildcard-cert   kubernetes.io/tls   3      6m5s

</pre>

**Comment by eigood** [comment]
This ended up being because the certificateRefs had group "core", instead of group "".  So not part of this original bug.  Sorry for the noise.

**Comment by timonwong** [comment]
Link to the PR: #39732
